### PR TITLE
Feature/improve dialog hosted window

### PR DESCRIPTION
**Pull request by ceo1647**
stays open behavior improvement

### CODE REVIEW

File: src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs [hunk 1]
```diff
@@ -21,28 +24,40 @@ protected DialogWindowHostedToolBase(IUIVisualizerService uiVisualizerService)
...
 
-        _windowViewModel = InitializeViewModel();
-        _windowViewModel.ClosedAsync += OnClosedAsync;
+        _parameter = parameter;
         ApplyParameter(_parameter);
 
         if (IsModal)
         {
-            await _uiVisualizerService.ShowDialogAsync(_windowViewModel, OnWindowCompleted);
+            _uiVisualizerService.ShowDialogAsync(_windowViewModel, OnWindowCompleted);
```
Comment (GeertvanHorrik) [inline src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs:47]: I think a modal task should always be awaited?

File: src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs [hunk 2]
```diff
@@ -21,28 +24,40 @@ protected DialogWindowHostedToolBase(IUIVisualizerService uiVisualizerService)
...
         }
         else
         {
-            await _uiVisualizerService.ShowAsync(_windowViewModel, OnWindowCompleted);
+            _uiVisualizerService.ShowAsync(_windowViewModel, OnWindowCompleted);
```
Comment (GeertvanHorrik) [inline src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs:51]: _ =

**Review by GeertvanHorrik — CHANGES_REQUESTED** [review]

File: src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs (hunk 1 above)
Comment (ceo1647) [reply src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs:47]: I need this operations not awaited
from here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs4014?f1url=%3FappId%3Droslyn%26k%3Dk(CS4014)
image

Comment (GeertvanHorrik) [reply src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs:47]: why would you not want to await a modal dialog?

Comment (ceo1647) [reply src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs:47]: because i want to finish this function, and keep tool open until window closed...

Comment (ceo1647) [reply src/Orc.Controls/Tools/DialogWindowHostedToolBase.cs:47]: i such case all tools works the same... tools with/without window logic

**Review by GeertvanHorrik — APPROVED** [review]

**Pull request merged**